lib/wab/impl/data.rb

```diff
@@ -415,29 +415,21 @@ def clone_value(value)
       def detect_hash(h)
         h.each_key { |k|
           v = h[k]
-          vc = v.class
-          if Hash == vc
-            detect_hash(v)
-          elsif Array == vc
-            detect_array(v)
-          elsif String == vc
-            v2 = detect_string(v)
-            h[k] = v2 if v2 != v
+          case v
+          when Hash   then detect_hash(v)
+          when Array  then detect_array(v)
+          when String then h[k] = detect_string(v) unless detect_string(v) == v
```

Comment:
Agreed. I'll place all the then statements in a new line for consistency
But notice how the branches are very similar except for the calling methods (`each_key` vs `each_index`)?